manager/allocator/networkallocator/portallocator.go

```diff
@@ -285,7 +285,8 @@ func (pa *portAllocator) portsAllocatedInHostPublishMode(s *api.Service) bool {
 
 	if s.Spec.Endpoint != nil {
```

Comment:
@aaronlehmann Sorry just got back to this now. Looking at the code I cannot find how we could end up with `Spec.Endpoint == nil && Endpoint != nil` for host mode published ports. Regardless, for the sake of this function, it would return `true` which is consistent with the hypothetical condition if ever it was to happen.